### DISCUSSION TITLE
Remove Old x86 'ibrew' Installation on Mac mini (M1).

### DISCUSSION THREAD

**Discussion by jasondh** [Homebrew/discussions#2952] [Getting started]
I wish to remove my old  x86 _'ibrew'_ installation on my M1 Mac under /usr/local/ - while keeping my standard ARM64 Home-brew installation in tact (/opt/homebrew/).

Can anyone confirm the correct way to do this please?  Cheers.

**Comment by gromgit** [comment]
This should do it:
```
arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```

**Reply by jasondh** [reply]
Thanks - do I need to open Terminal in Rosetta first - Or does it not matter?  Cheers.

**Reply by gromgit** [reply]
`arch -x86_64` forces execution of the Intel architecture portion of the selected universal binary (`/bin/bash` in this case). The uninstaller script that `bash` runs then checks the architecture it's running under (via `uname -m`) and nukes the appropriate hierarchy.

So no, it doesn't matter which architecture your terminal session is running in.

**Reply by jasondh** [reply]
Hi.  I tried the above, but the script wants to uninstall my **ARM64** installation /opt/homebrew/
file attached - [homebrew.txt](https://github.com/Homebrew/discussions/files/8091643/homebrew.txt)
Do I need to set --path /usr/local/Homebrew/ to the above script  somehow?

**Reply by cho-m** [reply]
Behavior is probably due to what I mentioned here: https://github.com/Homebrew/discussions/discussions/417#discussioncomment-1556606

Specifically, uninstaller prioritizes `brew` command on your `PATH`, so it will pick up ARM64 installation if `/opt/homebrew/bin` is on your `PATH`.

Examples you can try (just do one):
1. run script with `--path=/usr/local`. One way is to download copy and run locally, or you might be able to do one-liner (can't guarantee behavior since I haven't used):
    ```sh
    curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh | arch -x86_64 /bin/bash -s -- --path=/usr/local
    ```
2. modify your `PATH` to not have any `/opt/homebrew` or prioritize Rosetta, e.g. for latter,
    ```sh
    eval "$(/usr/local/bin/brew shellenv)"
    arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
    ```

**Reply by gromgit** [reply]
> Specifically, uninstaller prioritizes `brew` command on your `PATH`, so it will pick up ARM64 installation if `/opt/homebrew/bin` is on your `PATH`.

Ah, forgot about that quirk.

> 1\. run script with `--path=/usr/local`. One way is to download copy and run locally, or you might be able to do one-liner (can't guarantee behavior since I haven't used):
> ```shell
> curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh | arch -x86_64 /bin/bash -s -- --path=/usr/local
> ```

This works, but no longer asks you to confirm the uninstall, since stdin is no longer a terminal. Downloading and running the uninstaller separately is safer, if more verbose:
```
curl -fsSL -o /tmp/uninstall.sh https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh
arch -x86_64 /bin/bash /tmp/uninstall.sh --path=/usr/local
```

> 2\. modify your `PATH` to not have any `/opt/homebrew` or prioritize Rosetta, e.g. for latter,
> ```shell
> eval "$(/usr/local/bin/brew shellenv)"
> arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
> ```

This permanently changes the PATH of your current shell. While it's mostly harmless if the uninstall completes properly, it's just as easy to make the change temporary:
```
PATH=/usr/local/bin:$PATH arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```

**Comment by markddavidoff** [comment]
For anyone this helps, I specifically had to specify the x86 brew path  over at `/opt/homebrew/`
```
# not sure if this echo call was needed but, i wanted to be sure that $HOMEBREW_PREFIX was /opt/homebrew
echo $HOMEBREW_PREFIX #=> /opt/homebrew

curl -fsSL -o /tmp/uninstall.sh https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh
# this --path call is what i needed
arch -x86_64 /bin/bash /tmp/uninstall.sh --path=/opt/homebrew

# and then cleanup with
sudo rm -rf /opt/homebrew
```
